agent/engine/docker_task_engine.go

```diff
@@ -675,7 +703,24 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 			Error: CannotStartContainerError{errors.Errorf("Container not recorded as created")},
 		}
 	}
-	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+	dockerContainerMD := client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+
+	// Get metadata through container inspection and available task information then write this to the metadata file
+	// Performs this in the background to avoid delaying container start
+	// TODO: Add a state to the api.Container for the status of the metadata file (Whether it needs update) and
+	// add logic to engine state restoration to do a metadata update for containers that are running after the agent was restarted
+	if dockerContainerMD.Error == nil && engine.cfg.ContainerMetadataEnabled && !container.IsInternal() {
+		go func() {
+			err := engine.metadataManager.Update(dockerContainer.DockerID, task.Arn, container.Name)
+			if err != nil {
+				seelog.Warnf("Update metadata file failed for container %s of task %s: %v", container, task, err)
```

Comment:
minor: you can `return` here and get rid of `else`.

Reply:
fixed.